api/validator_test.go

```diff
@@ -12,25 +12,81 @@ import (
+	{input: models.Santander, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.fees not available for this bank"}]}`}},
+	{input: models.Citibank, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.fees not available for this bank"}]}`}},
+	{input: models.Itau, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.fees not available for this bank"}]}`}},
+	{input: models.JPMorgan, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.fees not available for this bank"}]}`}},
+	{input: models.Pefisa, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.fees not available for this bank"}]}`}},
+}
+
+var validationRulesFailedParametersV2 = []bankNumberParameter{
+	{input: models.BancoDoBrasil, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.rules not available for this bank"}]}`}},
+	{input: models.Santander, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.rules not available for this bank"}]}`}},
+	{input: models.Citibank, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.rules not available for this bank"}]}`}},
+	{input: models.Itau, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.rules not available for this bank"}]}`}},
+	{input: models.JPMorgan, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.rules not available for this bank"}]}`}},
+	{input: models.Pefisa, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.rules not available for this bank"}]}`}},
+}
+
+var validationRulesSuccessParametersV1 = []bankNumberParameter{
+	{input: models.BancoDoBrasil, expected: bankNumberExpectedParameter{code: 200}},
+}
+
+var validationRulesFailedParametersV1 = []bankNumberParameter{
+	{input: models.Caixa, expected: bankNumberExpectedParameter{code: 400, message: `{"errors":[{"code":"MP400","message":"title.rules not available in this version"}]}`}},
+}
+
 func Test_ValidateRegisterV1_WhenWithoutRules_PassSuccessful(t *testing.T) {
-	router, w := arrangeMiddlewareRoute("/validateV1", parseBoleto, validateRegisterV1)
-	body := test.NewStubBoletoRequest(models.BancoDoBrasil).WithExpirationDate(time.Now()).Build()
-	req, _ := http.NewRequest("POST", "/validateV1", bytes.NewBuffer([]byte(util.ToJSON(body))))
+	for _, fact := range validationRulesSuccessParametersV1 {
+		router, w := arrangeMiddlewareRoute("/validateV1", parseBoleto, validateRegisterV1)
+		body := test.NewStubBoletoRequest(fact.input).WithExpirationDate(time.Now()).Build()
```

Comment:
Para adicionar o Bradesco vc precisa incluir a wallet no BoletoRequest. Ela pode ser inserida para todos os banco pq para esse teste será invariante para os demais bancos.
```suggestion
	body := test.NewStubBoletoRequest(fact.input).WithExpirationDate(time.Now()).WithWallet(25).Build()
```

Reply:
done